src/state-manager/TransactionConsensus.ts

```diff
@@ -1024,6 +1094,565 @@ class TransactionConsenus {
         }
       }
     )
+
+    Comms.registerGossipHandler(
+      'poqo-receipt-gossip',
+      (payload: AppliedReceipt2) => {
+        profilerInstance.scopedProfileSectionStart('poqo-receipt-gossip')
+        try {
+          const queueEntry = this.stateManager.transactionQueue.getQueueEntrySafe(payload.txid)
+          if (queueEntry == null) {
+            /* prettier-ignore */ if (logFlags.error) this.mainLogger.error(`poqo-receipt-gossip no queue entry for ${payload.txid}`)
+            return
+          }
+          if (queueEntry.hasSentFinalReceipt === true) {
+            // We've already send this receipt, no need to send it again
+            return
+          }
+
+          if (logFlags.verbose) this.mainLogger.debug(`POQo: received receipt from gossip for ${queueEntry.logID} forwarding gossip`)
+
+          const executionGroupNodes = new Set(queueEntry.executionGroup.map((node) => node.publicKey))
+          const hasTwoThirdsMajority = this.verifyAppliedReceipt(payload, executionGroupNodes)
+          if (!hasTwoThirdsMajority) {
+            /* prettier-ignore */ if (logFlags.error) this.mainLogger.error(`Receipt does not have the required majority for txid: ${payload.txid}`)
+            nestedCountersInstance.countEvent('poqo', 'poqo-receipt-gossip: Rejecting receipt because no majority')
+            return
+          }
+          
+          queueEntry.poqoReceipt = payload
+          queueEntry.appliedReceipt2 = payload
+          queueEntry.recievedAppliedReceipt2 = payload
+          Comms.sendGossip(
+            'poqo-receipt-gossip',
+            payload,
+            null,
+            null,
+            queueEntry.transactionGroup,
+            false,
+            4,
+            payload.txid,
+            '',
+            true
+          )
+          queueEntry.hasSentFinalReceipt = true
+        } finally {
```

Comment:
no catch

Reply:
this may be fixed or removed in another branch.  created  GOLD-210 to verify.  Thanks!